common/src/main/java/com/zegelin/prometheus/cassandra/collector/StorageServiceMBeanMetricFamilyCollector.java

```diff
@@ -79,14 +79,18 @@ private StorageServiceMBeanMetricFamilyCollector(final StorageServiceMBean stora
         final Stream.Builder<MetricFamily> metricFamilyStreamBuilder = Stream.builder();
 
         {
-            final Stream<NumericMetric> ownershipMetricStream = storageServiceMBean.getOwnership().entrySet().stream()
-                    .map(e -> new Object() {
-                        final InetAddress endpoint = e.getKey();
-                        final float ownership = e.getValue();
-                    })
-                    .map(e -> new NumericMetric(metadataFactory.endpointLabels(e.endpoint), e.ownership));
-
-            metricFamilyStreamBuilder.add(new GaugeMetricFamily("cassandra_token_ownership_ratio", null, ownershipMetricStream));
+            try {
+                final Stream<NumericMetric> ownershipMetricStream = storageServiceMBean.getOwnership().entrySet().stream()
+                        .map(e -> new Object() {
+                            final InetAddress endpoint = e.getKey();
+                            final float ownership = e.getValue();
+                        })
+                        .map(e -> new NumericMetric(metadataFactory.endpointLabels(e.endpoint), e.ownership));
+
+                metricFamilyStreamBuilder.add(new GaugeMetricFamily("cassandra_token_ownership_ratio", null, ownershipMetricStream));
+            } catch (Throwable ex) {
+                logger.warn("Failed to get token ownership: {}", ex.getMessage());
```

Comment:
Probably best to use the `warn(...)` variant that takes a throwable, so we get a full stacktrace.
(https://www.slf4j.org/apidocs/org/slf4j/Logger.html#warn(java.lang.String,%20java.lang.Throwable))

Reply:
but thats the thing, you dont want that, do you ... the result would be completely same, the log when Cassandra starts but is not fully up and Prometheus is already reading /metrics, would be polluted by long stacktrace

Reply:
When logging exceptions and not rethrowing them, it's usually a good idea to log the whole thing so you get the stack trace and any cause(s).

The right fix should really to somehow avoid trying to collect while a node is still starting up. But defining the point at which a node has started is not easy.

Reply:
It would be then better to leave this as is if we are not after such a low-hanging fruit. This seems to me like relatively easy way how to get out of that. It is not like an user would not have any info when something goes wrong, there is still a message provided to him. I do not see any other places in that method try block when things could go wrong expect that one and we hit that one just once.